src/components/tools/Stats/StatOptions.svelte

```diff
@@ -39,17 +41,56 @@
     }
   }
 
+  /**
+   * When user types in the year, the combobox dropdown menu is filtered to show only
+   * values that complete the characters typed in the combobox
+   **/
+  function shouldFilterItem(item, value) {
+    if (!value) return true;
+    return item.text.includes(value);
+  }
+
+  /**
+   * When user presses the Enter or Tab key in combobox after typing in a year, update
+   * the combobox selected index if the year matches an item in the combobox dropdown menu
+   **/
+  function updateIndex(e) {
+    const { key, keyCode, target } = e;
+    if (["Enter", "Tab"].includes(key) || [13, 9].includes(keyCode)) {
+      const { value, id } = target;
+      switch (id) {
+        case "years-select-start":
+          startYear_selectedIndex = items.findIndex((d) => d.text === value);
+          return;
+        case "years-select-end":
+          endYear_selectedIndex = filteredItems.findIndex(
+            (d) => d.text === value
+          );
+          return;
+        default:
+          return;
+      }
+    }
+  }
+
   $: yearsList = range(dateRange[0], dateRange[1] + 1, 1);
```

Comment:
@mukhtyar these could be moved above the `function` declarations so that they're easier to discover when first reading the code.

Reply:
good idea. fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/191/commits/5a00c0d20c0db4938f704824e06a6611cba1425a